### DISCUSSION TITLE
nuget - support multi-targeting

### DISCUSSION THREAD

**Discussion by mieliespoor** [renovatebot/renovate#17915] [Request Help]
A common concept in the dotnet world is the ability to multi-target different frameworks. 

You will then see something like this in the csproj:

```xml
  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
    <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="3.1.12" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="3.1.18" />
  </ItemGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
    <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="6.0.7" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="6.0.7" />
  </ItemGroup>
```

How can I tell renovate that it shouldn't offer me net6.0 versions for netcoreapp3.1? If not using multi-targeting, this is very easy as I can use package rules along with `alloedVersions`.  Will `matchCurrentVersion` for the 3.1 versions to stick to that an not offer me any higher version?

**Comment by rarkins** [comment]
I think you can use `matchCurrentVersion` and `allowedVersions` in a package rule
